middleware: wifi_nxp: align imu adapter with SDK Next Generation

fengming-ye opened the pull request:
Update imu adapter to latest and
use cmake in imu_adapter folder instead of rpmsg.

dleach02 reviewed (changes requested):
@fengming-ye please realign this PR with the recent changes to integrated SDK-NG and move glue logic to Zephyr repo.

fengming-ye commented:
close PR due to code already synced

fengming-ye commented:
> @fengming-ye please realign this PR with the recent changes to integrated SDK-NG and move glue logic to Zephyr repo.

Hi @dleach02 
Updated. Old commits are not needed. Instead I updated wifi_nxp to align with zephyr change.
This PR is dependent by https://github.com/zephyrproject-rtos/zephyr/pull/88013
